engine/src/flutter/display_list/dl_op_records.h

```diff
@@ -484,8 +484,7 @@ struct TransformResetOp final : TransformClipOpBase {
 
 // 4 byte header + 4 byte common payload packs into minimum 8 bytes
 // DlRect is 16 more bytes, which packs efficiently into 24 bytes total
-// DlRoundRect is 48 more bytes, which rounds up to 48 bytes
-//         which packs into 56 bytes total
+// DlRoundRect is 52 more bytes, which packs into 60 bytes total
```

Comment:
These structs round up to a multiple of 8 bytes I believe which means that change will cause more padding to to occur. Look around for a size comment that talks about "rounding up" and borrow that wording.

Reply:
I anticipated 8-byte alignment too but I got these numbers by `printf` the `sizeof` (on macOS). Could this be platform specific?

Reply:
The struct is not 8-byte padded, but they are installed in the buffer on 8-byte alignments (see the templated Push method at the top of the file).

Also, there are 2 different ops that embed this object - Clip and Draw? And DrawDDRect maybe? Scan the file for other uses.

As a query tool - instead of printf, I sometimes use "static_assert(sizeof(foo) == N)" and then adjust N until my IDE tells me it isn't failing. Then I know the size. I also use this to plan how to lay out the fields so it takes fewer byes if it can. I don't think there is any leeway here to achieve that

And yes, sometimes it can be different on different platforms, but I've only seen that happen so far with vectors and shared_ptrs which have different layouts on Windows.

Reply:
And there is a practical reason for predicting the sizes of the objects - there is a test that makes a bunch of different DisplayLists and verifies that they concatenate to the expected size (i.e. testing alignment and such). Look in `dl_test_snippets.cc` for a bunch of "Invoker"s in a giant array that list the expected DL buffer sizes for each item it defines. You'll likely just have to add the delta to anything that uses RoundRect (and twice for the DRRect version).

The easiest way to debug this is `out/<engine variant>/display_list_unittests --gtest_filter="DisplayListTest.SingleOpSizes"` which will test each entry individually for appropriate sizes and print which entry in the giant array of test data is off. Once that passes, all of the combinatorial tests should also pass.